lib/src/tasks/coverage/api.dart

```diff
@@ -475,4 +631,45 @@ class CoverageTask extends Task {
       return port;
     }
   }
+
+  Future<List<int>> _test_functional(File file) async {
+
+    int count = observatoryPort.length;
+
+    String executable = 'pub';
+    List args = ['run',
+//        int port = await getOpenPort();
+//        String executable = 'dart';
+//        List args = ["--observe=${port}"
+    file.path.split(config.test.functionalTests[0] + '/')[1]];
+    print(config.test.functionalTests[0]);
+    TaskProcess process = new TaskProcess(executable, args,workingDirectory:config.test.functionalTests[0]);
+
+    String _observatoryFailPattern = 'Could not start Observatory HTTP server';
+    RegExp _observatoryPortPattern = new RegExp(
+        r'Observatory listening (at|on) http:\/\/127\.0\.0\.1:(\d+)');
+
+    String _testsFailedPattern = 'Some tests failed.';
+    RegExp _testsPassedPat = new RegExp(r'All\s[\d]*\s?tests passed.');
+
+    process.stderr.listen((l) => _coverageErrorOutput.add('    $l'));
+
+    await for (String line in process.stdout) {
+      _coverageOutput.add('    $line');
+      if (line.contains(_observatoryFailPattern)) {
+        throw new CoverageTestSuiteException(file.path);
```

Comment:
Most of the logic in this area is duplicated from how all other tests are run.. I think if we can move the dependency on selenium out of this task and run all tests via a pub serve process, then we could eliminate the need for this separate test execution & coverage collection path for functional tests